File: python/design-patterns/command/invoker/invoker.py
class RemoteControl:
    def __init__(self):
        self.history = []
    def execute_command(self, command):
        command.execute()
        self.history.append(command)
    def press_undo(self):
        if self.history:
            cmd = self.history.pop()
            cmd.undo()

File: typescript/design-patterns/command/invoker/Invoker.ts
// Invoker: Executed commands are stored in a history stack for undo support
class RemoteControl {
  private history: Command[] = [];

  executeCommand(command: Command) {
    command.execute();
    this.history.push(command);
  }

  undoButton() {
    const lastCommand = this.history.pop();
    if (lastCommand) {
      lastCommand.undo();
    }
  }
}